components/org.wso2.carbon.identity.application.authenticator.passive.sts/src/main/java/org/wso2/carbon/identity/application/authenticator/passive/sts/PassiveSTSAuthenticator.java

```diff
@@ -160,4 +173,33 @@ public String getFriendlyName() {
     public String getName() {
         return PassiveSTSConstants.AUTHENTICATOR_NAME;
     }
+
+    @Override
+    protected void initiateLogoutRequest(HttpServletRequest request, HttpServletResponse response,
+            AuthenticationContext context) throws LogoutFailedException {
+        ExternalIdPConfig externalIdPConfig = context.getExternalIdP();
+        String idpURL = context.getAuthenticatorProperties()
+                .get(IdentityApplicationConstants.Authenticator.PassiveSTS.IDENTITY_PROVIDER_URL);
+        String logOutPage;
+
+        try {
+            String replyUrl =
+                    IdentityUtil.getServerURL(FrameworkConstants.COMMONAUTH, true, true) + "?logOutKey=" + context
```

Comment:
Server url can be different than the load balanced endpoint that exposed to public. Check other authenticators, how it handled this.